### PR TITLE
prevent cron job running on forks

### PR DESCRIPTION
**Pull request by mmulholla**
Made it so the cron job will only run from the openshift-helm-charts/charts repository.

workflow dispatch is not restricted.

### CODE REVIEW

File: .github/workflows/schedule.yml [hunk 1]
```diff
@@ -25,18 +25,35 @@ jobs:
...
+          if [ $GITHUB_EVENT_NAME == 'workflow_dispatch' ]; then
+            echo '::set-output name=run-job::true'
+          elif [ $GITHUB_REPOSITORY == "openshift-helm-charts/charts" ]; then
+            echo '::set-output name=run-job::true'
+          else
+            echo '::set-output name=run-job::false'
+          fi
+
       - name: Install oc
+        if: steps.check_main.outputs.run-job == 'true'
```
Comment (zonggen) [inline .github/workflows/schedule.yml:43]: Think this should be `check_repo`, and the following `check_main`s.

**Comment by zonggen** [comment]
Can you rebase on top of latest 'main' branch?

**Review by zonggen — APPROVED** [review]
Tests passed and code LGTM!

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Pull request merged**